web/client/api/usersession/browser.js

```diff
@@ -15,13 +15,22 @@ export const getSessionName = (name) => "mapstore.usersession." +
  */
 export default {
     getSession: name => Observable.defer(() => {
-        const serialized = localStorage.getItem(getSessionName(name));
+        let serialized = {};
+        try {
+            serialized = localStorage.getItem(getSessionName(name));
+        } catch (e) {
+            console.error(e);
+        }
         const session = serialized && JSON.parse(serialized);
```

Comment:
In case of error you are going to parse {}, not a string.
I would just put all the code inside the try block, and just a return in the catch. What do you think?

Reply:
What is the code using getSession() expecting in case of error? Does it crash if we just return null, do we have to return an array like [0, {}] maybe?

Reply:
i think is safe to return a promise that resolves an array like [0, null]

in the end every error is catched [here](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/epics/usersession.js#L116)